### PR TITLE
Implemented visualizer

### LINKED ISSUE #28: Stateless Visualizer

### PR DESCRIPTION
**Pull request by angushtlam**
Closes #28 #44

# Description
This diff connects the navbar to run and step the code from the executor. The information from the interpreter will then sent to the visualizer state manager and rendered in the p5js sketch.

# Change Log
`src/view/components/white_button.js` renamed file for consistency, implemented onClick for component
`src/interpreter/executor/interpret.js`: exported functions for React app to use
`src/view/editor/index.js`: added read-only mode with preformatted code.
`src/view/index.js`: implemented interpreter state. added handler functions for subcomponent behaviors
`src/view/navbar/index.js`: added behavior to Run and Step buttons
`src/view/visualizer/sketch/index.js`: updated to not be a dummy sketch and use the newly introduced sketch state manager
`src/view/visualizer/state.js`: the new sketch state manager that handles incoming interpreter actions

# Manual Test Steps
The Run and Step buttons currently works like this
1. You can start program execution with either Run or Step.
2. Behavior differs between Run and Step here:
    - When you click on Run, the interpreter will automatically step through all steps one at a time at an interval of 1 second.
    - When you click on Step, you will begin the step

Clicking on Run while the interpreter is running will automatically step. Clicking on Run while the interpreter is running automatically will not do anything.

Clicking on Step while the interpreter is running automatically will stop the interpreter from automatically running, and this point on the user will need to step the rest of the way until the program ends.

When the program is running, the editor will be converted to a read-only mode. It will stay in read-only until the program is completed.

screen shot 2017-10-08 at 7 11 17 pm
screen shot 2017-10-08 at 7 11 27 pm

You can see the visualizer display the correct variables at each point of the program.

### CODE REVIEW

File: src/interpreter/executor/interpret.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ let isBlockComment = false;
 let userCode = []
 
 // Setup for when we initially check syntax
-function setup(inputCode) {
+export function setup(inputCode) {
```
Comment (zsmoore) [inline src/interpreter/executor/interpret.js]: I don't think this needs to be exported, it is never used.

**Comment by zsmoore** [comment]
Currently if you step and then run, the code hangs and will not do anything

**Comment by zsmoore** [comment]
Is there anyway to add line highlighting to this PR?  If we have comments or lines where nothing is interpreted the programming currently looks like it is hanging when it is actually just reading no data.

**Comment by angushtlam** [comment]
@zsmoore That is intended. You will need to step manually if you began stepping at any point.

**Comment by zsmoore** [comment]
@angushtlam I thought we had agreed run will just finish out the steps starting from where we are at?  Is there any downsides to switching to this functionality?

**Comment by zsmoore** [comment]
Can we get variable overflow functionality.  Since we don't shift, we just draw off the canvas.  Check for canvas bounds and shift if we are over it.

Check out https://github.com/zsmoore/Making-Art-With-Code/blob/master/100_things/sketch.js  
I do that here in the setup .

**Comment by noahpoz** [comment]
@zsmoore I can submit a separate PR for line highlighting

**Comment by zsmoore** [comment]
@noahpoz thats fine, you can just reference @angushtlam 's representation as you go along and do that then.

**Comment by angushtlam** [comment]
@zsmoore I think the variable overflow should go in a separate PR.

I also added the ability to run while in step in a new commit

**Comment by zsmoore** [comment]
@angushtlam can you check your commit again? When I press step now the program gets fully run.

**Comment by angushtlam** [comment]
@zsmoore Sorry, this commit should have better checks in place now.

**Pull request merged**